Assets/Scripts/Managers/SaveManager.cs

```diff
@@ -0,0 +1,166 @@
+        /// loads the savegame from the file & sets the states of gameobjects sequentially afterwards...
+        /// </summary>
+        /// <param name="fileName"></param>
+        public void Load(string fileName)
+        {
+            try
+            {
+                // get save object
+                Save save = Read(fileName);
+
+                // get game components
+                GameObject player = PlayerManager.Instance.GetPlayer();
+                InventoryController inventoryController = InventoryManager.Instance.GetInvController();
+                DayNightManager timeManager = DayNightManager.Instance;
+
+                // set player position
+                player.GetComponent<NavMeshAgent>().Warp(save.playerPosition);
+
+                // set player state
+                PlayerState state = player.GetComponent<PlayerState>();
+                state.SetHealth(save.playerHealth);
+                state.SetHydration(save.playerHydration);
+                state.SetSanity(save.playerSanity);
+                state.SetSaturation(save.playerSaturation);
+
+                // set time
+                timeManager.SetTimeOfDay(save.timeOfDay);
+                timeManager.SetDayNumber(save.dayNumber);
+
+                // set campsites and blueprints
+                List<SavedCampsite> savedCampsites = save.campsites;
+                savedCampsites.ForEach(cs =>
+                {
+                    cs.campsite.buildingBlueprints = new List<BuildingBlueprint>();
+                    cs.blueprints.ForEach(sbp =>
+                    {
+                        cs.campsite.buildingBlueprints.Add(sbp.blueprint);
+                        sbp.blueprint.Building = sbp.building;
+                        if (sbp.blueprintActive && sbp.buildingActive)
+                        {
+                            sbp.building.IsBuilt = true;
+                            sbp.blueprint.gameObject.GetComponentInChildren<BlueprintTooltip>().gameObject
+                                .SetActive(false);
+                            //sbp.blueprint.ShowBlueprint();
+                            //sbp.blueprint.ShowBuilding();
+                        }
+                        else if (sbp.blueprintActive && !sbp.buildingActive)
+                        {
+                            sbp.building.IsBuilt = false;
+                            //sbp.blueprint.ShowBlueprint();
+                        }
+                    });
+                    cs.SetState();
+                });
+
+                //set harvestables
+                save.harvestables.ForEach(h => { h.SetState(); });
+
+                // set inventory
+                inventoryController.SetItems(save.items);
+            }
+            catch (Exception e)
+            {
+                Debug.Log(e);
+            }
+        }
+
+
+        //JSON - Utility functions - interaction with File system...
+
+        private void Write(Save save, string fileName)
```

Comment:
I'd suggest adding an empty string as default for `fileName` for saving and loading :)
```suggestion
        private void Write(Save save, string fileName = "")
```

Reply:
kk